src/studio/src/designer/backend/ViewModels/Request/CreateDeploymentRequestViewModel.cs

```diff
@@ -22,5 +26,38 @@ public class CreateDeploymentRequestViewModel
         [Required]
         [JsonProperty("env")]
         public EnvironmentModel Environment { get; set; }
+
+        /// <summary>
+        /// Determines if this instance of the model is valid.
+        /// </summary>
+        /// <param name="validationContext">The current context of the validation check</param>
+        /// <returns>A list of validation results.</returns>
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
+            List<ValidationResult> issues = new List<ValidationResult>();
+            
+            if (string.IsNullOrEmpty(TagName))
+            {
+                issues.Add(new ValidationResult($"Tag name cannot be empty", new[] { nameof(TagName) }));
+                return issues;
+            }
+            
+            if (TagName[0] == '.' || TagName[0] == '-')
+            {
+                issues.Add(new ValidationResult($"Tag name cannot start with '.' or '-'.", new[] { nameof(TagName) }));
+            }
+
+            if (TagName.Length > 128)
+            {
+                issues.Add(new ValidationResult($"Tag name cannot be longer than 128 characters.", new[] { nameof(TagName) }));
+            }
+
+            if (!Regex.IsMatch(TagName, "^[a-z0-9.-]*$"))
```

Comment:
Underscore and upper case letters are also allowed in the specification.
"must be valid ASCII and may contain lowercase and uppercase letters, digits, underscores, periods and dashes. A tag name may not start with a period or a dash and may contain a maximum of 128 characters."

Reply:
I'll admit I simply copied the validation rules from the frontend code. I assumed that those rules were fine.

Reply:
Ok, then we are good. :)